pkg/connector/directmedia.go

```diff
@@ -119,6 +120,24 @@ func (wa *WhatsAppConnector) Download(ctx context.Context, mediaID networkid.Med
 			} else if err != nil {
 				return err
 			}
+
+			metadata := msg.Metadata.(*waid.MessageMetadata)
+			var directMediaMetadata *msgconv.FailedMediaKeys
+			if err := json.Unmarshal(metadata.DirectMediaMeta, &directMediaMetadata); err != nil {
+				return fmt.Errorf("failed to unmarshal message metadata %w", err)
+			}
```

Comment:
You don't need to re-decode the metadata, it was already decoded in the wrapping function (`var keys *msgconv.FailedMediaKeys`)